dvc/repo/metrics/show.py

```diff
@@ -190,7 +190,7 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
 
         if not typ and isinstance(o.metric, dict):
             t = o.metric.get(o.PARAM_METRIC_TYPE, typ)
```

Comment:
can we first write a shot summary of the semantics we would like to achieve? Including type and xpath. They both are important here. It seems it's not very straightforward now and is getting even more complicated with this change.

Reply:
@shcheklein what do you mean by that? Some kind of comment explainin what is the logic of extracting particular metric if xpath is defined?

Reply:
Yes, but it also obviously involves `type` argument as I can tell from the current implementation. For example, it decides on overriding `xpath` by taking a look at `type`. Do we want to keep it? Do we want to override `type` as well if `xpath` is defined? etc ... just want to see the documentation written in a holistic way for those too options :)

Reply:
Sure, just Ill try to prepare something tomorrow!